File: CSharp/BinaryTreeRightView.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<int> RightSideView(TreeNode root) {
        var res = new List<int>();
        if (root == null) { return res; }

        Helper(root, res, 0);
        return res;
    }

    private void Helper(TreeNode curr, IList<int> result, int currDepth) {
        if (curr == null) { return; }
        if (currDepth == result.Count)
        {
            result.Add(curr.val);
        }

        Helper(curr.right, result, currDepth + 1);
        Helper(curr.left, result, currDepth + 1);
    }
}

File: python/tree/binary_tree_right_view.py
from python.common import TreeNode


class RightSideView:
    def right_side_view(self, root: TreeNode | None) -> list[int]:
        result = []
        def visit(node, depth):
            if not node: return
            if depth == len(result): result.append(node.val)
            visit(node.right, depth + 1); visit(node.left, depth + 1)
        visit(root, 0)
        return result
